session.go

```diff
@@ -0,0 +1,113 @@
+package scylla
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"scylla-go-driver/transport"
+)
+
+// TODO: Should we support connecting to different than default port?
+// TODO: Add retry policy.
+// TODO: Add Query Paging.
+// TODO: make better randomConnection.
+
+type SessionConfig struct {
+	hosts []string
+	transport.ConnConfig
+}
+
+func MakeSessionConfig(tcpNoDelay bool, defaultConsistency uint16, hosts ...string) SessionConfig {
+	cpy := make([]string, len(hosts))
+	copy(cpy, hosts)
+
+	return SessionConfig{
+		hosts: cpy,
+		ConnConfig: transport.ConnConfig{
+			TCPNoDelay:         tcpNoDelay,
+			Timeout:            5 * time.Second, // Should user have control of timeout value?
+			DefaultConsistency: defaultConsistency,
+		},
+	}
+}
+
+// I don't know if we need this.
+func discoverHosts(hosts []string) []net.IP { // nolint:deadcode,unused // This might be used in the future.
+	var res []net.IP
+	for _, h := range hosts {
+		if ip, err := net.LookupIP(h); err != nil {
+			res = append(res, ip[0])
+		}
+	}
+	return res
+}
+
+type Session struct {
+	cfg     SessionConfig
+	cluster *transport.Cluster
+}
+
+func NewSession(cfg SessionConfig) (*Session, error) {
+	cluster, err := transport.NewCluster(cfg.ConnConfig, nil, cfg.hosts...)
+	if err != nil {
+		return nil, err
+	}
+
+	s := &Session{
+		cfg:     cfg,
+		cluster: cluster,
+	}
+
+	return s, nil
+}
+
+func (s *Session) NewQuery(content string) Query {
```

Comment:
How about `Query(cql string) *Query`?

Reply:
1. There is Query method which runs everything. What should I name it then? 
```go
func (s *Session) Query(req Query) (Result, error) {
	conn := s.randomConnection()
	if conn == nil {
		return Result{}, errNoConnection
	}

	res, err := conn.Query(req.stmt, nil)
	return Result(res), err
}
```
2. I just want to be sure if the pointer is going to point to query struct on stack, right?

Reply:
@mmatczuk ping